### DISCUSSION TITLE
Supplemental values in leaderboard submissions

### DISCUSSION THREAD

**Discussion by cIymax** [RetroAchievements/RAWeb#1632] [Ideas]
I recall I may have seen a suggestion similar to this before, but I can't seem to find it now.  Also, this discussion relates to rcheevos as well, but the rcheevos repository doesn't appear to have a Discussions section, so I decided to post here instead.

**Suggestion**
1. When making Leaderboards, it would be nice to be able to add an optional number of "Supplemental Value" groups.
2. This can be similar to how we can add an optional number of Alt Groups.
3. The supplemental values are submitted when the Value group's Measured is submitted.
4. The supplemental values do _not_ affect the ordering of entries in the leaderboards.  The ordering will still be based on the Value group's Measured.
5. We can limit how many supplemental values are supported, to start.  Even a max of 1 supplemental value can make a big difference.

**What This Allows**
1. Leaderboards can now show not only the ranking but also the supplemental info for each leaderboard entry.  The supplemental info can be interesting and/or informative to players.
2. With the supplemental info captured, RAWeb can optionally also be updated to enable dropdown filters on specific values of each type of supplemental info.
3. There's not a really good alternative to capture this supplemental info otherwise.  One approach is to make a separate, dedicated leaderboard for each value of a particular type of supplemental info (e.g., playable character).  But that approach is cumbersome if there is some way we can capture the same data in a single leaderboard.  That approach also doesn't scale well; one of the games I'm working on has 65 playable characters, which suggests 65 leaderboards.

**Use-Case Examples**

These examples are based on selectable character, but other types of info can be used, such as equipped weapon, hero level, lives lost, continues used, etc.
1. In a RTA speedrun leaderboard for a fighting game, the value will be the leaderboard timer, while the supplemental value can show which character you played as.  Example games: Street Fighter III: Third Strike, Ultimate Mortal Kombat 3.
5. In an IGT speedrun leaderboard for a JRPG or SRPG, the value will be the in-game timer, while the supplemental value can show which members you have in your party.  Example games: Final Fantasy X, Chrono Trigger, Final Fantasy Tactics, Romancing Saga.
6. In a RTA speedrun leaderboard for a 2D platformer, the value will be the leaderboard timer, while the supplemental value can show which character you played as.  Example game: Mega Man X4, Mega Man & Bass.
7. In a RTA speedrun leaderboard for a racing game, the value will be the leaderboard timer, while the supplemental value can show which character you played as.  Example game: Mario Kart 64.

**Comment by Tsearo** [comment]
I think I understand your idea but to clarify do you mean something like Sonic 2 could have a Speedrun split for Sonic or Tails based on this data and then have a single leaderboard? Would the idea be to show filtered views of the leaderboard for any character, Sonic only or Tails only then? Sounds like a nice idea for reducing clutter and giving nice overall data on which characters are best overall while still allowing individual character filtering if that is the idea

**Reply by cIymax** [reply]
Yes, Sonic 2 is a great example.  See my mockup below for what I had in mind.  It's exactly like you described.

![image](https://github.com/RetroAchievements/RAWeb/assets/94894395/643ae0b9-40f3-474b-bf67-5469b91f3c46)

**Reply by Tsearo** [reply]
Ah nice yeah that is what I was imagining you meant. Seems like it could be a great next step for leaderboard improvements

**Comment by suspect15** [comment]
I think is a fantastic idea and offers some great opportunities.  I can see this being very popular in racing game leaderboards to note which vehicle was used, high scores in games with different characters, etc.  Lots of potential value added use cases exist.
